### PR TITLE
Avoid more needless project dirtying

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , more work on insuring our project instance doesn't get flagged as dirty needlessly.

The goal is to be able to open the project properties, change _nothing_, click on [ OK ], and not have our project dirtied.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
image

Goal achieved, opening the project properties and clicking on OK without modifying any settings does _not_ dirty the project anymore.

@nyalldawson , relevant tests added.

**Comment by nyalldawson** [comment]
Nice nice!

I wonder if it's possible to add a test for the project properties dialog? You could try constructing a dialog, then closing it straight away and make sure the project isn't dirtied. That'd protect against future regressions here.

**Comment by nirvn** [comment]
@nyalldawson , something like this? I'm expecting it to fail on travis, looking for a gdb trace.

**Comment by nirvn** [comment]
@nyalldawson , OK, the test initially crashed because the map canvas overview is not created in the context of a test app. All good now.

File: src/core/qgsproject.cpp [hunk 1]
```diff
@@ -628,6 +635,9 @@ QgsCoordinateReferenceSystem QgsProject::crs() const
 
 void QgsProject::setCrs( const QgsCoordinateReferenceSystem &crs )
 {
+  if ( crs == mCrs )
```
Comment (nyalldawson) [inline src/core/qgsproject.cpp]: Could you add a unit test here?

Comment (nirvn) [reply src/core/qgsproject.cpp]: @nyalldawson , done.

**Pull request merged**